### ISSUE TITLE
No connected devices found

### ISSUE THREAD

**Issue by osben** [ionic-team/native-run#42]
**λ  native-run --list**
iOS
---
Connected Devices:
  No connected devices found
Virtual Devices:
  No virtual devices found
Android
-------
Connected Devices:
  No connected devices found
Virtual Devices:
  Pixel 2 API 28 (API 28) Pixel_2_API_28

**λ adb devices**
List of devices attached
abfe908d0504    device

**Comment by osben** [comment]
λ adb devices -l
List of devices attached
abfe908d0504           device product:tissot model:Mi_A1 device:tissot_sprout transport_id:1

**Comment by osben** [comment]
2019-06-08T21:49:17.277Z native-run Caught fatal error: Error: Command failed: C:\Users\osben\AppData\Local\Android\Sdk\platform-tools/adb -s abfe908d0504 shell getprop
[native-run]
[native-run]     at ChildProcess.exithandler (child_process.js:304:12)
[native-run]     at ChildProcess.emit (events.js:196:13)
[native-run]     at maybeClose (internal/child_process.js:1011:16)
[native-run]     at Process.ChildProcess._handle.onexit (internal/child_process.js:268:5) {

**Comment by borisjotic** [comment]
I had similar issue on macos (no devices), for me bug was solved by downgrading ionic-cli from 5.x to 4.12.0...
my ionic info (working one):
```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.4.2
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

Cordova:

   cordova (Cordova CLI) : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms     : ios 5.0.1
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.0.1, (and 4 other plugins)

System:

   ios-deploy : 1.9.4
   ios-sim    : 8.0.1
   NodeJS     : v12.4.0 (/usr/local/Cellar/node/12.4.0/bin/node)
   npm        : 6.9.0
   OS         : macOS Mojave
   Xcode      : Xcode 10.0 Build version 10A255
```

**Comment by dwieeb** [comment]
@osben I'm guessing this is related to https://github.com/ionic-team/native-run/issues/36

**Comment by Incubbus** [comment]
Same Problem.

**Comment by dwieeb** [comment]
@Incubbus Can you post the output of `native-run --list --verbose` ?

**Comment by Incubbus** [comment]
```
  client:usbmuxd connectUsbmuxdSocket +0ms
  client:usbmuxd getDevices +33ms
  protocol:usbmux socket write: {"messageType":"ListDevices"} +0ms
  native-run:android:utils:sdk:resolveSDKRoot Looking for $ANDROID_HOME +0ms
  native-run:android:utils:sdk:resolveSDKRoot Looking for $ANDROID_SDK_ROOT +1ms
  native-run:android:utils:sdk:resolveSDKRoot Looking at following directories: [ 'C:\\Users\\\\AppData\\Local\\Android\\Sdk' ] +2ms
  native-run:ios:list There was an error getting the iOS simulator list: { Error: spawnSync xcodebuild ENOENT
  native-run:ios:list     at Object.spawnSync (internal/child_process.js:982:20)
  native-run:ios:list     at Object.spawnSync (child_process.js:601:24)
  native-run:ios:list     at Object.getXcodeVersionInfo (C:\Users\\AppData\Roaming\npm\node_modules\native-run\dist\ios\utils\xcode.js:8:46)
  native-run:ios:list     at Object.getSimulators (C:\Users\\AppData\Roaming\npm\node_modules\native-run\dist\ios\utils\simulator.js:14:36)
  native-run:ios:list     at Object.list (C:\Users\\AppData\Roaming\npm\node_modules\native-run\dist\ios\list.js:19:43)
  native-run:ios:list     at Promise.all.Promise.resolve.then.then.iosList (C:\Users\\AppData\Roaming\npm\node_modules\native-run\dist\list.js:5:85)
  native-run:ios:list     at process._tickCallback (internal/process/next_tick.js:68:7)
  native-run:ios:list     at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
  native-run:ios:list     at startup (internal/bootstrap/node.js:283:19)
  native-run:ios:list     at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
  native-run:ios:list   errno: 'ENOENT',
  native-run:ios:list   code: 'ENOENT',
  native-run:ios:list   syscall: 'spawnSync xcodebuild',
  native-run:ios:list   path: 'xcodebuild',
  native-run:ios:list   spawnargs: [ '-version' ] } +0ms
  native-run:android:utils:sdk:resolveSDKRoot Using C:\Users\\AppData\Local\Android\Sdk +34ms
  native-run:android:utils:sdk:resolveEmulatorHome Looking for $ANDROID_EMULATOR_HOME +0ms
  native-run:android:utils:sdk:resolveEmulatorHome Using $HOME/.android/ at C:\Users\\.android +2ms
  native-run:android:utils:sdk:resolveAVDHome Looking for $ANDROID_AVD_HOME +0ms
  native-run Caught fatal error: { Error: No valid Android AVD home found.
  native-run     at resolveAVDHome (C:\Users\\AppData\Roaming\npm\node_modules\native-run\dist\android\utils\sdk\index.js:159:11) code: 'ERR_AVD_HOME_NOT_FOUND', exitCode: 1, data: undefined } +0ms
ERR_AVD_HOME_NOT_FOUND: No valid Android AVD home found.
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:27015
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1117:14)
Emitted 'error' event at:
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

**Comment by dwieeb** [comment]
@Incubbus I assume you took out your user directory in that output? or was this the actual path it tried? `C:\Users\\.android`?

Usually the AVD home is in your home folder at `~/.android/avd`. Does that directory exist? Have you opened Android Studio? If it doesn't, perhaps native-run should create it. This may be a premature error.

**Comment by Incubbus** [comment]
@dwieeb yes, i removed my user directory name.

There is no `avd`-folder because i am not using virtual devices for testing.

**Comment by dwieeb** [comment]
@Incubbus Thanks, I created https://github.com/ionic-team/native-run/issues/49 for that issue.

**Comment by Incubbus** [comment]
@dwieeb Creating the AVD folder may resolve one exception thrown, but does make the command finish without throwing another error...

```
native-run:android:utils:adb:getDeviceProperties Invoking adb: 'C:\\Users\\...\\AppData\\Local\\Android\\Sdk\\platform-tools\\adb' [ '-s', '...', 'shell', 'getprop' ] +0ms
  native-run:android:utils:adb:getDevices Found adb devices: [ { serial: '...',
  native-run:android:utils:adb:getDevices     state: 'device',
  native-run:android:utils:adb:getDevices     type: 'emulator',
  native-run:android:utils:adb:getDevices     connection: null,
  native-run:android:utils:adb:getDevices     properties:
  native-run:android:utils:adb:getDevices      { product: '...',
  native-run:android:utils:adb:getDevices        model: '...',
  native-run:android:utils:adb:getDevices        device: '...',
  native-run:android:utils:adb:getDevices        transport_id: '3' },
  native-run:android:utils:adb:getDevices     manufacturer: '... ...',
  native-run:android:utils:adb:getDevices     model: '...',
  native-run:android:utils:adb:getDevices     product: '...',
  native-run:android:utils:adb:getDevices     sdkVersion: '28' } ] +362ms
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:27015
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1117:14)
Emitted 'error' event at:
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

(Redacted with `...` again)

**Comment by pranaydutta89** [comment]
+1 same issue

**Comment by dwieeb** [comment]
Hey all, I just released native-run 0.2.7 which addresses @osben's original issue! 🎉 

Update: `npm i -g native-run@latest`

If there are more issues with device recognition on Android after 0.2.7, please create new issues. Thanks 😄

**Comment by davazro** [comment]
I can not run a basic tabs application generated with  ionic start myapp tabs on my device.  This is the log of  **native-run --list --verbose** command


```
PS D:\dav_mis_doc\cursos\Ionic 5 Udemy\3_10_tabs\myApp> native-run --list --verbose
  client:usbmuxd connectUsbmuxdSocket +0ms
  client:usbmuxd getDevices +24ms
  protocol:usbmux socket write: {"messageType":"ListDevices"} +0ms
  native-run:android:utils:sdk:resolveSDKRoot Looking for $ANDROID_HOME +0ms
  native-run:android:utils:sdk:resolveSDKRoot Looking for $ANDROID_SDK_ROOT +10ms
  native-run:android:utils:sdk:resolveSDKRoot Looking at following directories: [ 'C:\\Users\\Usuario\\AppData\\Local\\Android\\Sdk' ] +0ms
  native-run:android:utils:sdk:resolveSDKRoot Using C:\Users\Usuario\AppData\Local\Android\Sdk +33ms
  native-run:android:utils:sdk:resolveEmulatorHome Looking for $ANDROID_EMULATOR_HOME +0ms
  native-run:android:utils:sdk:resolveEmulatorHome Looking at $HOME/.android +0ms
  native-run:android:utils:sdk:resolveEmulatorHome Using $HOME/.android/ at C:\Users\Usuario\.android +2ms       
  native-run:android:utils:sdk:resolveAVDHome Looking for $ANDROID_AVD_HOME +0ms
  native-run:android:utils:sdk:resolveAVDHome Looking at $HOME/.android/avd +1ms
  native-run:android:utils:sdk:resolveAVDHome Using $HOME/.android/avd/ at C:\Users\Usuario\.android\avd +1ms    
  native-run:android:utils:sdk:getSDKPackage Parsing C:\Users\Usuario\AppData\Local\Android\Sdk\platform-tools\package.xml +0ms
  native-run:android:utils:avd:getAVDINIs Discovered AVD ini files: [ 'C:\\Users\\Usuario\\.android\\avd\\Nexus_4_API_22.ini',
  native-run:android:utils:avd:getAVDINIs   'C:\\Users\\Usuario\\.android\\avd\\Nexus_5_API_26.ini' ] +0ms       
  native-run:android:utils:adb:getDevices Invoking adb: 'C:\\Users\\Usuario\\AppData\\Local\\Android\\Sdk\\platform-tools\\adb' [ 'devices', '-l' ] +0ms
  native-run:android:utils:sdk:findAllSDKPackages Walking C:\Users\Usuario\AppData\Local\Android\Sdk to discover 
SDK packages +0ms
  native-run:android:utils:adb:parseAdbDevices Parsing adb devices from output lines: [ 'List of devices attached',
  native-run:android:utils:adb:parseAdbDevices   '407ccc9a               device product:lavender_eea model:Redmi_Note_7 device:lavender transport_id:14',
  native-run:android:utils:adb:parseAdbDevices   '',
  native-run:android:utils:adb:parseAdbDevices   '' ] +0ms
  native-run:android:utils:adb:getDeviceProperties Invoking adb: 'C:\\Users\\Usuario\\AppData\\Local\\Android\\Sdk\\platform-tools\\adb' [ '-s', '407ccc9a', 'shell', 'getprop' ] +0ms
  native-run:android:utils:adb:getDevices Found adb devices: [ { serial: '407ccc9a',
  native-run:android:utils:adb:getDevices     state: 'device',
  native-run:android:utils:adb:getDevices     type: 'hardware',
  native-run:android:utils:adb:getDevices     connection: null,
  native-run:android:utils:adb:getDevices     properties:
  native-run:android:utils:adb:getDevices      { product: 'lavender_eea',
  native-run:android:utils:adb:getDevices        model: 'Redmi_Note_7',
  native-run:android:utils:adb:getDevices        device: 'lavender',
  native-run:android:utils:adb:getDevices        transport_id: '14' },
  native-run:android:utils:adb:getDevices     manufacturer: 'Xiaomi',
  native-run:android:utils:adb:getDevices     model: 'Redmi Note 7',
  native-run:android:utils:adb:getDevices     product: 'lavender_eea',
  native-run:android:utils:adb:getDevices     sdkVersion: '28' } ] +409ms
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:27015
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1104:14)
Emitted 'error' event at:
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
